### PR TITLE
Adds new security laser sprites and flavor examine descriptions

### PR DESCRIPTION
**Pull request by Stokes52**
## What Does This PR Do
Adds new item and in-hand sprites for the laser gun and energy.
Adds new flavorful descriptions when examined, for the LORE!

guns_comparison

I've tried to strike a balance between being similar to the old silhouette for gameplay/familiarity while also giving a sleek and updated look.

The new flavor descriptions are just for fun. One references the (intentional) design similarities between the laser and the WT-550 security rifle, and the other references an existing in-game corporation, because more lore is always great.

## Why It's Good For The Game
Updates the very bland looking sprites of the two most common weapons and adds interesting flavor text to make using these weapons feel more immersive.  References existing lore and adds some lore.

## Images of changes
Laser
laser_description
laser_gun
laser_hand

Energy
energy_description
energy_gun
energy_hand

Animated!
guns_empty

guns_full

## Changelog
:cl:
add: New laser gun sprite + in hand sprite
add: New energy gun sprite + in hand sprite
add: New laser gun flavor description
add: New energy gun flavor description
fix: The tesla revolver ammo sprite was broken. I changed a 1 to a 0 so it appears again
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	ammo_x_offset = 3
 	flight_x_offset = 15
 	flight_y_offset = 10
+	shaded_charge = 1
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/nuclear.dm:13]:
```suggestion
	shaded_charge = TRUE
```

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 2]
```diff
@@ -30,6 +31,7 @@
 	charge_sections = 3
 	can_flashlight = 0 // Can't attach or detach the flashlight, and override it's icon update
 	actions_types = list(/datum/action/item_action/toggle_gunlight)
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/nuclear.dm:34]:
```suggestion
```

The `mini` sprite has charge states as well.

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 3]
```diff
@@ -51,6 +53,7 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode/hos, /obj/item/ammo_casing/energy/laser/hos, /obj/item/ammo_casing/energy/disabler)
 	ammo_x_offset = 4
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/nuclear.dm:56]:
```suggestion
```

Ditto

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 4]
```diff
@@ -79,6 +82,7 @@
 	can_flashlight = 0
 	trigger_guard = TRIGGER_GUARD_NONE
 	ammo_x_offset = 2
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/nuclear.dm:85]:
```suggestion
```

Same

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 5]
```diff
@@ -92,3 +96,4 @@
 	ammo_x_offset = 1
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/laser, /obj/item/ammo_casing/energy/disabler)
 	selfcharge = 1
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/nuclear.dm:99]:
```suggestion
```

Same

File: code/modules/projectiles/guns/energy/stun.dm [hunk 6]
```diff
@@ -15,7 +15,7 @@
 	origin_tech = "combat=4;materials=4;powerstorage=4"
 	ammo_type = list(/obj/item/ammo_casing/energy/shock_revolver)
 	can_flashlight = 0
-	shaded_charge = 1
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/stun.dm:18]:
```suggestion
```

File: code/modules/projectiles/guns/energy/stun.dm [hunk 7]
```diff
@@ -24,6 +24,7 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/disabler)
 	origin_tech = "combat=4"
 	ammo_x_offset = 2
+	shaded_charge = 0
```
Comment (dearmochi) [inline code/modules/projectiles/guns/energy/stun.dm:27]:
```suggestion
```

**Review by dearmochi — COMMENTED** [review]
A few things, also is `laser` supposed to be constantly blinking?

**Comment by Stokes52** [comment]
> A few things, also is `laser` supposed to be constantly blinking?

It follows the design of the "wt550" aka security auto rifle which also blinks.   HOWEVER, if that's too distracting, it can be removed to not blink. I was torn on it too.

@dearmochi 
Regarding the additions of shaded_charge = 0.  Without that added to the other guns, their ammo sprites will break.  Since I modified the energy gun to be shaded_charge = 0 (or shaded_charge = TRUE), it will break anything that is using the energy gun as a base.

The tesla revolver was one I discovered to be broken during my testing as well so I fixed it.

**Review by dearmochi — APPROVED** [review]

**Comment by S34NW** [comment]
These look superb, good work

**Review by S34NW — APPROVED** [review]

**Comment by E-MonaRhg** [comment]
Okay so first of all, amazing sprites, these all look great, good work. But I got _some points_ to make.

1. It might just be me, but the way the energy gun power indicator cuts the middle "block" when it is at 50% power looks a bit meh, something like this (behold, my non-existent spriting abilities) would _maybe_ look better.
image
2. The "golden shield of Nanotrasen Security" thing MIGHT be incorrect as the "sliver-kite shield with an engraved sword—of the Nanotrasen security forces" is what is on the CC officer berets, unless you are making a new, _normal security shield_ then fair enough.
image

3. This one is really minor because lore but, I don't think Warp-Tac Inc. exists within ours, but if you're making your own then more power to you and this point will be irrelevant. And Shellguard Munitions apparently specialises in anti-synthetic weapon systems and platforms, but I mean they still can make normal energy guns I guess, so once again irrelevant.

**Comment by LightFire53** [comment]
> 2. The "golden shield of Nanotrasen Security" thing MIGHT be incorrect as the "sliver-kite shield with an engraved sword—of the Nanotrasen security forces" is what is on the CC officer berets, unless you are making a new, _normal security shield_ then fair enough.

I'm pretty sure that if you inspect the sec uniforms (and especially the corporate security uniforms), it references the golden shield. CC security would be different of course because at that point you are an elite officer, not just a mall cop.

**Comment by E-MonaRhg** [comment]
> I'm pretty sure that if you inspect the sec uniforms (and especially the corporate security uniforms), it references the golden shield. CC security would be different of course because at that point you are an elite officer, not just a mall cop.

All security jumpsuits have some flavor of 
`"It's made of a slightly sturdier material than standard jumpsuits, to allow for robust protection."`

And the beret has
`"A beret with the security insignia emblazoned on it. For officers that are more inclined towards style than safety."`

Closest to would be the security jacket
`"A sturdy black jacket with reinforced fabric. Bears insignia of NT corporate security."`

I don't see any reference to "the golden shield" anywhere, but hey, maybe its a good moment to start having it.

**Comment by Stokes52** [comment]
At the moment the golden shield is meant to match the insignias/color scheme found on the corporate security hat, the HOS beret, and the security jacket.  Generally speaking gold is the accent color of choice for security which is why I chose to go that route. 

gold

The main point of the blue/gold is functional however. No other weapon has the golden shield making it immediately distinguishable from its peers.

**Comment by Stokes52** [comment]
After thinking about it, I decided to remove the green flashing light so that they both have consistent behavior:
Green = good
Red blinking = no ammo.

I also fixed the flashlight sprite offset so it sits on the railing properly.

update1

This PR should be ready to go.

**Comment by McRamon** [comment]
Overall shape is good, however grip on laser gun just screams "ballistics" while the gun is energy (i don't think energy gun has any recoil). Which is not good, gun doesn't feel like cheap laser rifle anymore. A bit better in terms of quality than current, but not great either.

**Comment by Stokes52** [comment]
Ergonomics is still important for aim and comfort, regardless of recoil. :)

**Review by Necaladun — APPROVED** [review]
@McRamon 's comments make sense, I'd like them to have more of a "cheap mass produced" feel because NT are cheap, however these are so much nicer than the others still I'd approve them as is, but feel free to tweak and re-poke me.

**Review by variableundefined — APPROVED** [review]

**Pull request merged**